std/file.d

```diff
@@ -3902,6 +3903,15 @@ string tempDir() @trusted
             DWORD len = GetTempPathW(buf.length, buf.ptr);
             if (len) cache = toUTF8(buf[0 .. len]);
         }
+        else version(Android)
+        {
+            // Don't check for a global temporary directory as
+            // Android doesn't have one.  Instead, use the local
+            // directory or set cache here to your app's local
+            // directory.
+
+            cache = getcwd();
```

Comment:
If you want your program to work multi-platform including Android and use a different temporary directory than the working directory, better just to not use `tempDir` than editing Phobos.